laura/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,47 +14,69 @@
 
 package com.google.sps;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 import java.util.Collection;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    ArrayList<TimeRange> answer = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> answer = new ArrayList<>();
 
     int lastOkStartTime = -1;
-    int intvervalLength = 0;
+    int maxOptionalAttendees = -1;
 
     for (int hour = 0; hour <= 23; hour ++) {
         for (int minute = 0; minute <= 59; minute ++) {
             int start = hour * 60 + minute;
             TimeRange potential = TimeRange.fromStartDuration(start, (int)request.getDuration());
+
+            if (!TimeRange.WHOLE_DAY.contains(potential)) 
+                break;
+
             boolean isOk = true;
+            for (String req_attendee : request.getAttendees())
+                isOk &= canAttend(events, potential, req_attendee);
```

Comment:
Here and in other places: even if it's just a one-liner, it would be better to surround the code with curly braces: 

```java
for (String req_attendee : request.getAttendees()) {
                isOk &= canAttend(events, potential, req_attendee);
}
```